order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java

```diff
@@ -70,10 +86,161 @@
+		}
+
+		return new ResponseEntity<>(orderMapper.toWS(order.get(), new CycleAvoidingMappingContext()), HttpStatus.OK);
+	}
+
+	@GetMapping(params = "userUuid")
+	@RequiresJwtAuthentication
+	@RequiresPermissions("order:read")
+	public ResponseEntity<?> readOrdersByUserUuid(@RequestParam(value = "userUuid") String userUuid) {
+
+		Optional<List<OrderBean>> tmpOrderBeans = orderService.findByUserUuid(userUuid);
+
+		if(!tmpOrderBeans.isPresent()){
+			log.error("Error in read orders. Fail in finding user uuid={}", userUuid);
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.USER_UUID_NOT_FOUND.name(), "Cannot read orders by user uuid that is not found"), HttpStatus.NOT_FOUND);
+		}
+
+		List<OrderBean> orderBeans = tmpOrderBeans.get();
+
+		List<OrderBean> newOrderBeans = orderBeans.stream().filter(orderBean ->
+			!orderBean.getOrderStatus().equals(OrderStatus.DELETED)
+		).collect(Collectors.toList());
+
+		return new ResponseEntity<>(orderMapper.toWSs(newOrderBeans, new CycleAvoidingMappingContext()), HttpStatus.OK);
+	}
+
+	@GetMapping(params = "agentUuid")
+	@RequiresJwtAuthentication
+	@RequiresPermissions("order:read")
+	public ResponseEntity<?> readOrdersByAgentUuid(@RequestParam(value = "agentUuid") String agentUuid) {
+
+		Optional<List<OrderBean>> tmpOrderBeans = orderService.findByAgentUuid(agentUuid);
+
+		if(!tmpOrderBeans.isPresent()){
+			log.error("Error in read orders. Fail in finding user uuid={}", agentUuid);
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.USER_UUID_NOT_FOUND.name(), "Cannot read orders by agent uuid that is not found"), HttpStatus.NOT_FOUND);
+		}
+
+		List<OrderBean> orderBeans = tmpOrderBeans.get();
+
+		List<OrderBean> newOrderBeans = orderBeans.stream().filter(orderBean ->
+				!orderBean.getOrderStatus().equals(OrderStatus.DELETED)
+		).collect(Collectors.toList());
+
+		return new ResponseEntity<>(orderMapper.toWSs(newOrderBeans, new CycleAvoidingMappingContext()), HttpStatus.OK);
+	}
+
+	/******************     NON API     *******************/
+	public ResponseEntity<?> validateUserAndAgentUuid(OrderWS orderWS) {
+
+		Optional<UserBean> userBean = accountService.findByUuid(orderWS.getUserUuid());
+		Optional<UserBean> agentBean = accountService.findByUuid(orderWS.getAgentUuid());
+
+		// check user uuid
+		if(!userBean.isPresent()){
+			log.error("Error in create order. User uuid={} not found", orderWS.getUserUuid());
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.USER_UUID_NOT_FOUND.name(), "In create order user uuid={" + orderWS.getUserUuid() + "} not found"), HttpStatus.NOT_FOUND);
+		}
+
+		// check agent uuid
+		if(!agentBean.isPresent()){
+			log.error("Error in create order. Agent uuid={} not found", orderWS.getAgentUuid());
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.AGENT_UUID_NOT_FOUND.name(), "In create order agent uuid={" + orderWS.getAgentUuid() + "} not found"), HttpStatus.NOT_FOUND);
+		}
+
+		// check user role
+		if(!userBean.get().isUser()) {
```

Comment:
shouldn't need to check this. as everyone is implicitly an end user.

Reply:
```
@PostMapping("/signup")
	public ResponseEntity<?> signup(@RequestParam(required = false, defaultValue = "USER") String role, @Validated @RequestBody UserWS userWS) {
		BindException errors = new BindException(userWS, "UserWS");

		signupValidator.validate(userWS, errors);

		if (errors.hasErrors()) {
			log.error("Error in signup user. Fail in signup validation fields. userWS={}", userWS.toString());
			List<ErrorWS> errorWSList = apiErrorsExceptionHandler.generateErrorWSList(errors);
			return new ResponseEntity<>(new ApiErrorsWS(errorWSList), HttpStatus.UNPROCESSABLE_ENTITY);
		}

		userWS.setRoles(Lists.newArrayList(role));

		if (accountService.userExist(userWS.getEmailAddress())) {
			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.ACCOUNT_EXIST.name(), "Account already registered."), HttpStatus.CONFLICT);
		}

		UserBean userBean = userMapper.toBean(userWS);

		UserBean user = accountService.createUser(userBean);

		try {
			emailService.sendVerificationTokenEmail(user.getEmailAddress(), user.getToken(), EmailType.ACTIVATION.name());
		} catch (MessagingException e) {
			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.EMAIL_MESSAGING_EXCEPTION.name(), e.getMessage()), HttpStatus.INTERNAL_SERVER_ERROR);
		}
		return new ResponseEntity<>(userMapper.toWS(user), HttpStatus.CREATED);
	}
```

Should I make the request params as a list and default value = user?

Reply:
Currently it is only specific role for each created Person

Reply:
Fixed